File: C/contains_duplicate(II).c
/* Given an array of integers and an integer k, find out whether there are two distinct indices i and j in the array such that nums[i] = nums[j] and the difference between i and j is at most k.
*/

bool containsNearbyDuplicate(int* nums, int numsSize, int k) {
    bool flag = false;
    if (nums == NULL || numsSize < 0 || k < 0) return flag;
    for (int i = 0; i < numsSize; i++) {
        for (int j = i+1; j < numsSize; j++) {
            if (nums[i] == nums[j] && (j-i<= k)) {
                flag = true;
            }
        }
    }
    return flag;
}

File: Python/contains_duplicate(II).py
""" Given an array of integers and an integer k, find out whether there are two distinct indices i and j in the array such that nums[i] = nums[j] and the difference between i and j is at most k.
"""
def containsNearbyDuplicate(self, nums, k):
    # type nums: List[int]
    # type k: int
    # rtype: bool
    array = {}
    if len(nums) == 0:
        return False
    for i in range(0, len(nums)):
        index = array.get(nums)
        if index != None and i - index <= k:
            return True
        else:
            array[nums[i]] = i
    return False

